### ISSUE TITLE
Anaconda installation fails due to post install scripts

### ISSUE THREAD

**Issue by katiyarsahil** [ContinuumIO/anaconda-issues#8856]
I am trying to reinstall anaconda 64 bit on Windows and the installations works right till the end but it fails executing the post install script and returns a Winerror 2: File not found error The system cannot find the file specified.
However, If i check the LIb folder in Anaconda3 folder, i see the files _nsis.py and subprocess.py which the installer says are missing.
Ignoring the error results in no anaconda navigator in the start menu but anaconda promt is available for some reason.

I tried installing Miniconda too but am facing the same issue.

Kindly help

![python error](https://user-images.githubusercontent.com/26725671/37369584-24aa79fa-26cf-11e8-9c9e-94f0354f7d19.png)

**Comment by daxiagithub** [comment]
I have the same issue when I was installing "Anaconda3-5.1.0-Windows-x86_64.exe" on Windows 7 (see the attached pictures).   Following the advice from Anaconda online forum, I tried turning off McAfee virus scan, but it didn't help.    Help please!

![condainstallationerror1](https://user-images.githubusercontent.com/14238547/37683820-018024ee-2c4b-11e8-9e5a-466acf4173d8.PNG)

![condainstallationerror2](https://user-images.githubusercontent.com/14238547/37683812-f5865f82-2c4a-11e8-8a8d-83b2abc760ce.PNG)

**Comment by katiyarsahil** [comment]
None of the fixes listed on various portals workewd for me to solve this issue. I was able to install Anaconda using previous Ananconda version (4.3.3) rather than Anaconda 5.1. kindly check the Ananconda version archives to get access to the previously released versions. https://repo.continuum.io/archive/

**Comment by extendann** [comment]
I am also getting the same error, please help me to resolve

**Comment by daxiagithub** [comment]
I just used the 4.3.3 version as suggested.  But this issue should not be
closed.

**Comment by kmnhz** [comment]
I am facing the same issue. The program seems to be full of bugs.

**Comment by blessy2018** [comment]
I also have the same problem please help...

**Comment by amitsharma1998** [comment]
I am having the same problem please help

**Comment by daxiagithub** [comment]
@amitsharma1998  and others,   Did you try the latest 5.2 release?  I haven't.

**Comment by surutusa** [comment]
Same issue here... My previous installation start failing some days ago. First activate command was not recognized by Anaconda Prompt. After uninstalling and reinstalling anaconda, I have been having this issue (with 5.2 release). I'm gonna try previous releases.

**Comment by pal-soumya** [comment]
I am also facing the issue, please help!!

**Comment by kmnhz** [comment]
I have installed Jupyter notebook through pip install.........Could not find any solution so far.

**Comment by mingwandroid** [comment]
Use conda?

**Comment by elferherrera** [comment]
After several failed installations with different version I think I found the solution. The problem is that the installation script is looking for cmd.exe, which in newer versions of windows is no longer in C:/Windows/system32. In order for the installation to work you need to copy cmd.exe from c;/windows/syswow64 and place it in the previous folder. Don't try to add the syswow64 path to your external variables, it doesn't work. The anaconda installation wont find it. 

I think the fact that Windows is moving from cmd.exe to powershell is a sign that Anaconda should be doing the same. Don't you think so?

**Comment by mingwandroid** [comment]
> The problem is that the installation script is looking for cmd.exe, which in newer versions of windows is no longer in C:/Windows/system32.

Sorry, this is not the case, if you do not have cmd.exe in C:\Windows\System32 then your system is broken.

> where cmd
> C:\Windows\System32\cmd.exe

**Comment by mingwandroid** [comment]
To explain this in more detail, `SysWOW64` in an abbreviation of System-Windows32-on-Windows64. The binaries in there are 32-bit binaries which are used to run such 'legacy' applications. `C:\Windows\System32` contains 64-bit binaries and should be preferred in all cases. As I said, if your 64-bit cmd.exe is missing then something is badly wrong with *your* system. Copying the 32-bit binary here is *not* an appropriate fix for this problem.

**Comment by elferherrera** [comment]
You are right. I'm checking with the IT guys to see what they did to my computer.

**Comment by SyedUmerHasan** [comment]
I am also having same problem i dont know why its happening, i deleted the last version of anaconda and install a new version that is causing this problem

**Comment by elferherrera** [comment]
In the error window that you get, python is telling you in which line number of subprocess.py is the error. The good thing is that even if the installation fails, the files are copied to your installation folder. Look for the subprocess.py script and find the line where with the error. That way I found that I was missing the cmd.exe from my computer.

**Comment by kanika1611** [comment]
Hi,
I am trying to install Anaconda in my office laptop and made sure it was white-listed. I ran the installation file as admin and i still get the error below:
![image](https://user-images.githubusercontent.com/36744474/52070139-2cc07d00-254e-11e9-86fb-6d18e93e1821.png)
I have a 64 bit system and my cmd is a 32 bit app.

Any leads will be appreciated! Thanks!

**Comment by csoja** [comment]
@kanika1611 If you are running into permissions errors ("Access is denied") - you should trying installing it for "Just Me".  There are many other tips to be found in the documentation: https://docs.anaconda.com/anaconda/install/windows/

**Comment by kanika1611** [comment]
Hi @csoja! 
Thanks. I tried that already. Doesn't seem to work :(

**Comment by MsFakhra** [comment]
same error. Is anybody able to resolve?

**Comment by Daniel-Sutton** [comment]
Line 1172 of subprocess.py is creating the process to run post_install.bat from the command line interpreter (cmd.exe).
In order to execute the command, the program find the full path of cmd.exe from the Windows system variable called ComSpec.
In Windows, ComSpec points to the command line interpreter and its value is normally : C:\WINDOWS\system32\cmd.exe . 
Anaconda3 rev. 2018.12 will fail if the value of ComSpec is wrong.  In that case ComSpec needs to be modified manually.  In Windows the environmental system variables are easily accessible and editable from the control panel...

**Comment by kanika1611** [comment]
Hi @Daniel-Sutton :+1: I checked the Comspec path and its set to C:\WINDOWS\system32\cmd.exe

**Comment by Daniel-Sutton** [comment]
Hi Kanika1611,
This is interesting..  This was a hard show stopper for me.  At the same time, I am not so surprised that the program subprocess.py could fail at line 1172 for other additional reasons.

I don’t know your level of expertise for programming but, I can tell you what I did to fix my problem.  By the way, I am not a Python developer.  I use it for my hobby.  I am just generally comfortable with programming.

After multiple installations, Python rev 3.7 was active on my computer.  I was able to open the command line interpreter to execute subprocess.py. 

C:\users\Daniel(Our your user..)\Anaconda3\Lib\  Python subprocess.py

Prior to execute subprocess.py, I edited it to print the value of some variables and the result of some commands before the execution of line 1172.  The print results are displayed on the command line interpreter window.  It was easy to figure out  what was wrong from the results of the prints.

VSCode was also available on my computer to edit subprocess.py and be able to locate line 1172.

You can add those lines just before line 1172 to reveal what can be wrong.:
print('comspec =', comspec)
print('executable = ', executable)
print('args =', args)
print('int(not close fds) = ', int(not close fds))
print('env =', env)
print('os.fspath(cwd) if cwd is not None else None,startupinfo = ', os.fspath(cwd) if cwd is not None else None,startupinfo)

**Comment by va26** [comment]
@Daniel-Sutton I tried printing the variables but it won't show any output. It doesnt give an error while running but it won't show any ouput. I tried placing these statements in different places but nothing seems to work :/

**Comment by Daniel-Sutton** [comment]
Have you tried to insert a print statement such as print(‘Hello’) at the very first line of subprocess.py ?
Before the line : # subprocess - Subprocesses with accessible I/O streams

Not running into errors when executing ..\python suprocess.py from a ‘command prompt window ’ ,  would means that python is responding and that it finds subprocess.py with no problem.  But, then it should show the print outputs and subprocess.py should crash and issue errors, unless everything is fixed.  So there is something wrong.
  
Make sure that python is operational and that you can output something from a print statement on the command prompt windows.

**Comment by ManideepPotluru** [comment]
I'm facing the same problem too ..idk what to do :(

**Comment by SyedUmerHasan** [comment]
this happens when you install 32 bit Anaconda on 64 bit machine, this
happened with me and took my 2 weeks to solve this problem...
try installing python / anaconda with same 32bit/64 bit respectively

**Comment by AgentNo** [comment]
I too am having the same problem. This is my first time trying to install Anaconda 3 (64-bit) on a 64 bit machine. I have tried everything suggested in this issue thread, as well as anything else I could scarper from Google and Stack Overflow. If anyone finds a solution, please help!

**Comment by daviesalex** [comment]
Data point: I hit this because COMSPEC was set to powershell.exe (this is a fairly common config for Windows users to force their default shell to be more modern). Not sure if its easy to avoid this, but I just removed the COMSPEC env variable to get the installer to run.

**Comment by AgentNo** [comment]
How exactly did you do that if you don't mind me asking.

**Comment by daviesalex** [comment]
@AgentNo there is a control panel widget "Edit the system environment variables" (I see it if I type "env" in the start menu). At the bottom of the window that opens is a 'Environment Variables' box and in there I just deleted COMSPEC from the user variable (where it was set). The system default at the bottom shows what it is set to.

**Comment by AgentNo** [comment]
@daviesalex That... actually worked. Thank you so much!

The installation now completes, but I can't find the Navigator anywhere on the system. It's not in the start menu and Windows Search can't find it.
